webapp/channels/src/components/common/hooks/useCWSAvailabilityCheck.ts

```diff
@@ -7,15 +7,20 @@ import {useSelector} from 'react-redux';
 import {Client4} from 'mattermost-redux/client';
 import {getConfig} from 'mattermost-redux/selectors/entities/general';
 
-export default function useCWSAvailabilityCheck() {
-    const [canReachCWS, setCanReachCWS] = useState(false);
+export default function useCWSAvailabilityCheck(teamEditionDefaultAllowed = false) {
+    const [canReachCWS, setCanReachCWS] = useState<boolean | undefined>(undefined);
     const config = useSelector(getConfig);
     const isEnterpriseReady = config.BuildEnterpriseReady === 'true';
     useEffect(() => {
         if (!isEnterpriseReady) {
             return;
         }
-        Client4.cwsAvailabilityCheck().then(() => setCanReachCWS(true));
+        Client4.cwsAvailabilityCheck().then(() => {
+            setCanReachCWS(true);
+
+            // server will respond with 400 for Team Edition. If teamEditionDefaultAllowed is true, and we have a 400
+            // Set availability to true, so trial requests from team edition can still be initiated in-product
+        }).catch((err) => setCanReachCWS(err.status_code === 400 && teamEditionDefaultAllowed));
```

Comment:
```suggestion
        }).catch((err) => setCanReachCWS(err.status_code === 400 && teamEditionAllowedToCheckAvailability));
```

Reply:
IMO the meaning of the value gets lost with that name change. The value doesn't determine whether or not team edition should be allowed to check availability (as by this point, it's already done so) - it defaults the availability to true if it is team edition (which we know by the 400 status)